### PR TITLE
swego: init at 0.7

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Swiss army knife Webserver in Golang. Similar to the Python
SimpleHTTPServer but with many features.

https://github.com/nodauf/Swego

Related to #81418

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/swego/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    owner = "nodauf";
+    repo = "Swego";
+    rev = "v${version}";
+    sha256 = "14qww4ndvrxmrgkggr8hyvfpz2v7fw0vq6s8715mxa28f8pfi78b";
+  };
+
+  vendorSha256 = "068drahh0aysrm8cr5pgik27jqyk28bsx5130mc2v3xd0xmk8yp1";
+
+  buildPhase = ''
+    go build -ldflags="-s -w" -o $pname src/webserver.go
```
Comment (SuperSandro2000) [inline pkgs/servers/swego/default.nix:20]: Please do not overwrite buildPhase. Please use buildFlagsArray.

File: pkgs/servers/swego/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
```
Comment (SuperSandro2000) [inline pkgs/servers/swego/default.nix:24]: Please use postInstall to fix binary names.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review. If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108106` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>swego</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 108106` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>swego</li>
  </ul>
</details>

```
building
# SimpleHTTPServer-golang/src/controllers
src/controllers/parsingArgs.go:130:4: undefined: PrintEmbeddedFiles
src/controllers/parsingArgs.go:138:4: undefined: PrintEmbeddedFiles
src/controllers/parsingArgs.go:144:4: undefined: PrintEmbeddedFiles
src/controllers/parsingArgs.go:153:2: undefined: PrintEmbeddedFiles
src/controllers/parsingArgs.go:162:2: undefined: PrintEmbeddedFiles
```

**Comment by kalbasit** [comment]
@grahamc build swego

**Comment by SuperSandro2000** [comment]
@fabaff ping

File: pkgs/servers/swego/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/servers/swego/default.nix]: Removed

File: pkgs/servers/swego/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/servers/swego/default.nix]: Doing the renaming now in `postInstall`.

Thanks for the feedback.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 108106` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>swego</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Should we exclude this package from darwin?

**Comment by SuperSandro2000** [comment]
> Should we exclude this package from darwin?

It could be an darwin issue but you can just set it broken for darwin.

**Comment by fabaff** [comment]
OK, marked as broken on darwin.

File: pkgs/servers/swego/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    description = "Simple Webserver in Golang";
+    longDescription = ''
+      Swiss army knife Webserver in Golang. Similar to the Python
+      SimpleHTTPServer but with many features.
+    '';
+    homepage = "https://github.com/nodauf/Swego";
+    license = with licenses; [ gpl2Only ];
+    maintainers = with maintainers; [ fab ];
+    # There is an issue with darwin
+    broken = !stdenv.isDarwin;
```
Comment (SuperSandro2000) [inline pkgs/servers/swego/default.nix:34]:
```suggestion
    broken = stdenv.isDarwin;
```

Comment (fabaff) [reply pkgs/servers/swego/default.nix:34]: Thanks, @SuperSandro2000 


Note to myself: Not very smart to set `NIXPKGS_ALLOW_BROKEN=1` while working on packages.

File: pkgs/servers/swego/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  meta = with lib; {
+    description = "Simple Webserver in Golang";
+    longDescription = ''
+      Swiss army knife Webserver in Golang. Similar to the Python
+      SimpleHTTPServer but with many features.
+    '';
+    homepage = "https://github.com/nodauf/Swego";
+    license = with licenses; [ gpl2Only ];
+    maintainers = with maintainers; [ fab ];
+    # There is an issue with darwin
```
Comment (SuperSandro2000) [inline pkgs/servers/swego/default.nix:33]:
```suggestion
    # darwin crashes with:
    # src/controllers/parsingArgs.go:130:4: undefined: PrintEmbeddedFiles
```

**Pull request merged**